### PR TITLE
Update ibc go simapp v8 used in CI

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #4009 

## Description

This PR updates the ibc-go simapp version used in CI from `v8.2.0` to `v8.3.1`.
This PR also fixes some typos in other changelog entries.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] ~Added tests: integration (for Hermes) or unit/mock tests (for modules).~
- [x] Linked to GitHub issue.
- [ ] ~Updated code comments and documentation (e.g., `docs/`).~ 
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Review by romac — APPROVED** [review]

File: tools/test-framework/src/chain/ext/ica.rs [hunk 1]
```diff
@@ -92,7 +94,54 @@ pub fn register_interchain_account<Chain: ChainHandle, Counterparty: ChainHandle
...
+        TaggedChannelId<Chain, Counterparty>,
+        TaggedPortId<Chain, Counterparty>,
+    ),
+    Error,
+> {
+    let wallet = chain.wallets().relayer().cloned();
+
+    let owner = handle.get_signer()?;
+
+    let version_str = format!("{{\"version\":\"ics27-1\",\"encoding\":\"proto3\",\"tx_type\":\"sdk_multi_msg\",\"controller_connection_id\":\"{}\",\"host_connection_id\":\"{}\"}}", connection.connection_id_a.0, connection.connection_id_b.0);
```
Comment (romac) [inline tools/test-framework/src/chain/ext/ica.rs:126]: Probably best to use the `serde_json::json!()` macro here.

File: tools/test-framework/src/chain/ext/ica.rs [hunk 2]
```diff
@@ -92,7 +94,54 @@ pub fn register_interchain_account<Chain: ChainHandle, Counterparty: ChainHandle
...
+
+    let tm = TrackedMsgs::new_static(vec![msg_any], "RegisterInterchainAccount");
+
+    let events = handle
+        .send_messages_and_wait_commit(tm)
+        .map_err(Error::relayer)?;
+
+    for event in events.iter() {
+        if let IbcEvent::OpenInitChannel(open_init) = &event.event {
+            let channel_id = open_init.channel_id.clone().ok_or(()).map_err(|_| Error::generic(eyre!("channel_id is empty in the event response after sending MsgRegisterInterchainAccount")))?;
```
Comment (romac) [inline tools/test-framework/src/chain/ext/ica.rs:144]: This seems like an overly long line, would have expected rustfmt to shorten it.

Comment (ljoss17) [reply tools/test-framework/src/chain/ext/ica.rs:144]: This seems to be ignored by `cargo fmt --all`, not sure why

File: tools/test-framework/src/chain/ext/ica.rs (hunk 1 above)
Comment (ljoss17) [reply tools/test-framework/src/chain/ext/ica.rs]: done in 1f6d019b067c98f4b0e7938b1f940d6496920218

File: tools/test-framework/src/chain/ext/ica.rs (hunk 2 above)
Comment (ljoss17) [reply tools/test-framework/src/chain/ext/ica.rs]: Manually done in 1f6d019b067c98f4b0e7938b1f940d6496920218

**Pull request merged**